internal/cltest/cltest.go

```diff
@@ -608,21 +607,23 @@ func CreateJobRunViaWeb(t *testing.T, app *TestApplication, j models.JobSpec, bo
 	err := ParseJSONAPIResponse(resp, &jr)
 	require.NoError(t, err)
 
-	assert.Equal(t, j.ID, jr.JobID)
+	assert.Equal(t, j.ID, jr.JobSpecID)
 	return jr
 }
 
 // CreateHelloWorldJobViaWeb creates a HelloWorld JobSpec with the given MockServer Url
 func CreateHelloWorldJobViaWeb(t *testing.T, app *TestApplication, url string) models.JobSpec {
 	j := FixtureCreateJobViaWeb(t, app, "../internal/fixtures/web/hello_world_job.json")
-	j.Tasks[0] = NewTask("httpget", fmt.Sprintf(`{"url":"%v"}`, url))
+	temp := NewTask("httpget", fmt.Sprintf(`{"url":"%v"}`, url))
+	j.Tasks[0].Params = temp.Params
```

Comment:
This looks a little nuanced, what changed to make this necessary? I think the NewTask factory was helpful when the `type` parameter was mixed in, but we might not need it anymore. You could possibly just set params with:
```
j.Tasks[0].Params = JSONFromString(fmt.Sprintf(`{"url":"%v"}`, url))
```

Reply:
If you simply add `j.Tasks[0] = NewTask(...)` you will end up adding a new task. Why? Because you don't explicitly delete the old `Task[0]` from the db and just save a new row in the db. This change modifies the existing task 0.